src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -179,8 +179,11 @@ def load_arguments(self, _):
                    help='Role of the service principal.')
         c.argument('skip_assignment', arg_type=get_three_state_flag(),
                    deprecate_info=c.deprecate(target='--skip-assignment', hide=True), help='No-op.')
-        c.argument('show_auth_for_sdk', options_list='--sdk-auth', deprecate_info=c.deprecate(target='--sdk-auth'),
-                   help='output result in compatible with Azure SDK auth file', arg_type=get_three_state_flag())
+        c.argument('show_auth_in_json', options_list=['--sdk-auth', '--json-auth'],
+                   deprecate_info=c.deprecate(target='--sdk-auth'),
+                   help='Output service principal credential along with cloud endpoints in JSON format. '
+                        'See https://learn.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli.',
```

Comment:
Personally, I prefer not to include this URL. People who don't know what `--sdk-auth` does most likely don't need to use it as well. 🤣